File: Heap/Kth Largest Element in an Array/findKthLargest.java
import java.util.PriorityQueue;

class findKthLargest{
    public int findKthLargest(int[] arr, int k){
        //heap to store the k largest elements
        PriorityQueue<Integer> pq = new PriorityQueue<>();

        for(int i=0; i<arr.length; i++){
            if(i<k){
                pq.add(arr[i]); //add the first k elements to the heap
            } else{
                //if the current element is greater than the smallest in the heap(root)
                if(arr[i] > pq.peek()){
                    pq.poll(); //remove the smallest element
                    pq.add(arr[i]); //dd the current element
                }
            }
        }

        return pq.peek(); //return the root (kth largest element)
    }

    public static void main(String[] args){
        findKthLargest sol = new findKthLargest();

        int[] arr = {3,2,1,5,6,4};
        int k = 3;

        int ans = sol.findKthLargest(arr, k);

        System.out.println(ans);
    }
};

File: Heap/Kth Largest Element in an Array/findKthLargest.cpp
#include <iostream>
#include <queue>
#include <vector>
using namespace std;

class Solution{
public:
    int finfKthLargest(vector<int>& arr, int k){
        //make a  (min-heap) priority queue where smallest element is at the top
        priority_queue<int, vector<int>, greater<int>> pq;

        for(int i=0; i<arr.size(); i++){
            if(i<k){
                pq.push(arr[i]); //add the first k elements to the heap
            } else{
                //if the current element is greater than the smallest in the heap
                if(arr[i] > pq.top()){
                    pq.pop(); //remove the smallest element
                    pq.push(arr[i]); //add the current element
                }
            }
        }

        return pq.top(); //return the root (kth largest element)
    }

    int main(){
        Solution sol;

        vector<int> arr = {3,2,1,5,6,4};
        int k = 3;

        int ans = sol.finfKthLargest(arr, k);

        cout<<ans<<endl;

        return 0;
    }
};